src/core-connector-svc/BaseRoutes.ts

```diff
@@ -0,0 +1,112 @@
+    SDKClientInitiateTransferError,
+    SDKNoQuoteReturnedError,
+} from '../domain/SDKClient';
+
+export class BaseRoutes {
+    protected handleResponse(data: unknown, h: ResponseToolkit, statusCode: number = 200) {
+        return h.response(data as ResponseValue).code(statusCode);
+    }
+
+    protected handleError(error: unknown, h: ResponseToolkit) {
+        if (error instanceof InvalidAccountNumberError) {
+            return h.response({ status: '3101', message: 'Invalid Account Number provided' }).code(400);
+        } else if (error instanceof AccountVerificationError) {
+            return h
+                .response({
+                    status: '3200',
+                    message: error.message,
+                })
+                .code(500);
+        } else if (error instanceof UnSupportedIdTypeError) {
+            return h
+                .response({
+                    status: '3100',
+                    message: 'Unsupported Id Type Error',
+                })
+                .code(500);
+        } else if (error instanceof FineractWithdrawFailedError) {
+            return h.response({ status: '4000', message: 'Fineract Withdrawal Error' }).code(500);
+        } else if (error instanceof SearchFineractAccountError) {
+            return h.response({ status: '3200', message: 'Search Fineract Account Error' }).code(500);
+        } else if (error instanceof FineractAccountNotFoundError) {
+            return h.response({ status: '3200', message: 'Fineract Account Not Found' }).code(404);
+        } else if (error instanceof FineractGetAccountWithIdError) {
+            return h
+                .response({ status: '2000', message: 'Error encountered while getting fineract account by ID' })
+                .code(500);
+        } else if (error instanceof FineractAccountNotActiveError) {
+            return h.response({ status: '4000', message: 'Fineract Account not active' }).code(500);
+        } else if (error instanceof FineractGetClientWithIdError) {
+            return h.response({ status: '4000', message: 'Getting client by ID encountered error' }).code(500);
+        } else if (error instanceof FineractDepositFailedError) {
+            return h.response({ status: '4000', message: 'Fineract Deposit Failed' }).code(500);
+        } else if (error instanceof FineractGetChargesError) {
+            return h.response({ status: '4000', message: 'Fineract Get charges error ' }).code(500);
+        } else if (error instanceof SDKNoQuoteReturnedError) {
+            return h.response({ status: '3200', message: 'SDK Client No Quote Returned' }).code(500);
+        } else if (error instanceof SDKClientContinueTransferError) {
+            return h.response({ status: '2000', message: 'SDK Client continue transfer error' }).code(500);
+        } else if (error instanceof SDKClientInitiateTransferError) {
+            return h.response({ status: '2000', message: 'SDK Client initiate transfer error' }).code(500);
```

Comment:
here is a lot of DRY violation in this code, and the logic of defining error messages and code should not be implemented here. It should be a part of custom error classes